context/app/static/js/components/entity-search/MetadataMenu/hooks.js

```diff
@@ -23,4 +25,18 @@ function useMetadataMenu(lcPluralType) {
   return { createNotebook, selectedHits, closeMenu };
 }
 
-export { useMetadataMenu };
+function useDatasetsAccessLevel(ids) {
+  const query = {
+    query: {
+      bool: {
+        must: [getIDsQuery(ids), getTermClause('mapped_data_access_level.keyword', 'Protected')],
+      },
+    },
+    _source: ['mapped_data_access_level'],
+    size: ids.length,
+  };
+  const { searchHits: datasets } = useSearchHits(query);
+  return { datasets };
+}
+
+export { useMetadataMenu, useDatasetsAccessLevel };
```

Comment:
Configured api response to remove unnecessary iteration of datasets for finding Protected datasets. This will directly give the aggregated count of Protected datasets.